### PR TITLE
Support OpretFirst (opret1st)

### LINKED ISSUE #6: Support OpretFirst in create_rgb20
When creating a contract genesis, the OwnedRights defaults to a TapretFirst revealed seal. See here:
https://github.com/BP-WG/bp-core/blob/c3ce5268c09afe823b1ca2ee3ef5f56cec483b4c/seals/src/txout/blind.rs#L91

When calling `into_seal_value_map` within `create_rgb20`, the above from method is called and there seems to be no way to override this behavior:
https://github.com/RGB-WG/rust-rgb20/blob/7599f0faa0397b89750b820920a0ec82079bea15/src/create.rs#L72

I believe this causes complications further down the line when attempting to transfer; I'm trying to debug why I get a different derived scriptpubkey hash when trying to construct a PSBT to transfer the token.

### PR DESCRIPTION
**Pull request by oacdutra**
Closes #6

### CODE REVIEW

File: src/create.rs [hunk 1]
```diff
@@ -49,6 +50,7 @@ impl<'consignment> Rgb20<'consignment> for Contract {
         precision: u8,
         allocations: OutpointValueVec,
         inflation: OutpointValueMap,
+        method: String,
```
Comment (dr-orlovsky) [inline src/create.rs]: Why the method comes as string and not enum?

File: Cargo.toml [hunk 2]
```diff
@@ -44,3 +44,7 @@ default = []
 all = ["serde", "cli"]
 serde = ["serde_crate", "serde_with", "lnpbp/serde", "bitcoin/use-serde", "rgb-std/serde", "amplify/serde", "chrono/serde"]
 cli = ["clap", "serde", "serde_yaml", "serde_json"]
+
+[patch.crates-io]
+# TODO: Remove after merge https://github.com/RGB-WG/rgb-std/pull/8
+rgb-std = { version = "0.8.0", git="https://github.com/crisdut/rgb-std/", branch="fix/opret1st" }
```
Comment (dr-orlovsky) [inline Cargo.toml]: Can you please put this into a separate commit which can be removed before merging?

**Review by dr-orlovsky — CHANGES_REQUESTED** [review]
Looks good apart from two comments. Can you pls also specify in the PR description that it depends on the other PR in rgb-stdlib?

**Comment by zoedberg** [comment]
@crisdut I've tried this PR (along with https://github.com/RGB-WG/rgb-std/pull/8) and setting OpretFirst as the close method when creating the RGB20 seems to work good

But reading https://github.com/RGB-WG/rust-rgb20/issues/6 I'm not really sure this PR was needed.

> I believe this causes complications further down the line when attempting to transfer; I'm trying to debug why I get a different derived scriptpubkey hash when trying to construct a PSBT to transfer the token.

@cryptoquick Could you please explain why do you think creating contracts using tapret should cause complications? In rgb-lib we issue contracts with tapret and then use opret to transfer them without any complications.

**Comment by oacdutra** [comment]
> @crisdut I've tried this PR (along with [RGB-WG/rgb-std#8](https://github.com/RGB-WG/rgb-std/pull/8)) and setting OpretFirst as the close method when creating the RGB20 seems to work good
> 
> But reading #6 I'm not really sure this PR was needed.
> 
> > I believe this causes complications further down the line when attempting to transfer; I'm trying to debug why I get a different derived scriptpubkey hash when trying to construct a PSBT to transfer the token.
> 
> @cryptoquick Could you please explain why do you think creating contracts using tapret should cause complications? In rgb-lib we issue contracts with tapret and then use opret to transfer them without any complications.


Hi @zoedberg, 

I think today is possible to make operations with differents deterministic commitments because the rgb-node does not check/verify the proof after created.

I found some codes for verifying proofs, but I'm not sure they are used.

Example:

https://github.com/BP-WG/bp-core/blob/master/dbc/src/anchor.rs#L423

**Comment by zoedberg** [comment]
Hi @crisdut,

> I found some codes for verifying proofs, but I'm not sure they are used.

I've checked with rgb-lib and can confirm that they are unused.

> I think today is possible to make operations with differents deterministic commitments because the rgb-node does not check/verify the proof after created.

This seems to me a considerable bug that needs a dedicated issue. IMO the correct way to proceed with this is to keep this PR open until we fix verification. Once that will work as expected we'll see if  this PR is really needed.

I've opened the issue: https://github.com/RGB-WG/rgb-node/issues/213

**Comment by oacdutra** [comment]
> Hi @crisdut,
> 
> > I found some codes for verifying proofs, but I'm not sure they are used.
> 
> I've checked with rgb-lib and can confirm that they are unused.
> 
> > I think today is possible to make operations with differents deterministic commitments because the rgb-node does not check/verify the proof after created.
> 
> This seems to me a considerable bug that needs a dedicated issue. IMO the correct way to proceed with this is to keep this PR open until we fix verification. Once that will work as expected we'll see if this PR is really needed.
> 
> I've opened the issue: [RGB-WG/rgb-node#213](https://github.com/RGB-WG/rgb-node/issues/213)

Hi @zoedberg,

Yes, totally agree

**Comment by zoedberg** [comment]
@dr-orlovsky this needs to be merged and then included in a 0.8 release. It's not a feature, as I initially thought, but it's a fix instead, necessary since https://github.com/RGB-WG/rgb-std/pull/9 has been merged

File: Cargo.toml (hunk 2 above)
Comment (dr-orlovsky) [inline Cargo.toml]: With the upstream released we now need to remove this git dependency - cargo wouldn't publish a new version without it. With that done I will be able to merge and releae

**Review by dr-orlovsky — CHANGES_REQUESTED** [review]
We need to update ti the released rgbstd and then I will be able to publish the update

File: Cargo.toml (hunk 2 above)
Comment (oacdutra) [reply Cargo.toml]: done!

**Comment by oacdutra** [comment]
> @dr-orlovsky this needs to be merged and then included in a 0.8 release. It's not a feature, as I initially thought, but it's a fix instead, necessary since [RGB-WG/rgb-std#9](https://github.com/RGB-WG/rgb-std/pull/9) has been merged

Hi @zoedberg , my bad!

I should have mentioned the relation between changes.

**Comment by zoedberg** [comment]
> Hi @zoedberg , my bad!
> 
> I should have mentioned the relation between changes.

No worries :)

**Review by dr-orlovsky — APPROVED** [review]
ACK 280badc286553a5652da2444e715dcae7685f99f
